tagstudio/src/core/library.py

```diff
@@ -2315,3 +2175,340 @@ def sort_fields(self, entry_id: int, order: list[int]) -> None:
+                is_selected: bool = True
+                # Remap entry.fields from list of dicts, to dict for ease of work
+                entry_fields: dict = self.remap_fields(entry)
+                for key, value in query_part.items():
+                    if isinstance(value, str):
+                        value = value.strip().casefold()
+                    entry_value = entry_fields.get(key)
+
+                    # special treatment for filename
+                    if key == "filename" and isinstance(value, str):
+                        is_selected = is_selected and self.check_filename(entry, value)
+                    # all usual tags handling, as if search was in AND mode
+                    elif key == "tag_id":
+                        if not self.handle_tag_id(value, entry_tags):
+                            is_selected = False
+                            break
+                    elif key == "unbound":
+                        entry_tuple = None
+                        # HACK
+                        if isinstance(query_words, str):
+                            query_words = [query_words]
+                        if isinstance(query_words, list):
+                            entry_tuple = self.handle_unbound(
+                                entry,
+                                all_tag_terms,
+                                entry_tags,
+                                entry_authors,
+                                query_words,
+                            )
+                        if entry_tuple is None:
+                            is_selected = False
+                            break
+                    elif key == "EMPTY":
+                        empty_fields = value
+                        if not isinstance(empty_fields, list):
+                            empty_fields = [empty_fields]
+                        if not self.required_fields_empty(entry_fields, empty_fields):
+                            is_selected = False
+                            break
+                    else:
+                        negative: bool = False
+                        if key[0] == "-":
+                            entry_value = entry_fields.get(key[1:])
+                            negative = True
+                        if not self.handle_common_field(entry_value, value, negative):
+                            is_selected = False
+                            break
+
+                if is_selected and entry_tuple is None:
+                    filtered_entries.append((ItemType.ENTRY, entry.id))
+                elif is_selected and entry_tuple is not None:
+                    filtered_entries.append(entry_tuple)
+            pre_results.append(filtered_entries)
+
+        # Entries should match all parts separated by '|'
+        result_set: set = set()
+        if search_mode == SearchMode.AND:
+            if len(pre_results) == 1:
+                return list(pre_results[0])
```

Comment:
since we know the `pre_results` has 1 item, and it's a list, it could be enough to return the value as it is, no?
```suggestion
                return pre_results
```